File: TestesAuto/Bild/Compra/CompraSteps.java
package bild.Compra;

import bild.Utils;
import bild.Obra.ObraPage;
import cucumber.api.java.en.And;
import cucumber.api.java.en.Given;
import cucumber.api.java.en.When;

public class CompraSteps {

	CompraPage pageCompra = new CompraPage();

	@Given("acessei o menu Compras >> Calendario de Compras")
	public void acessarMenuComprasCanlendarioDeCompras() throws InterruptedException {
		String nomeDaObra = ObraPage.getVariableYouWantToUse();
		pageCompra.acessarMenuComprasCanlendarioDeCompras(nomeDaObra);
		Utils.logPrint("Compra","Acessando o menu Compras");
	}

	@And("na tela de Calendario eu seleciono a Obra")
	public void selecionoObraEInsumos() throws InterruptedException {
		pageCompra.selecionoObraEInsumos();
		Utils.logPrint("Compra","Acessando o menu Calendario de Compras");
	}

	@And("na tela de Calendario eu clico em Comprar")
	public void clicarEmComprar() throws InterruptedException {
		pageCompra.clicarEmComprar();
		Utils.logPrint("Compra","na tela de Calendario eu clico em Comprar");
	}

	@When("na tela de Comprar Insumos eu clico em comprar saldo de todos insumos")
	public void clicarEmComprarSaldoTotal() throws InterruptedException {
		pageCompra.clicarEmComprarSaldoTotal();
		Utils.logPrint("Compra","Comprar saldo de todos insumos");
	}

	@And("na tela de ordem de compra eu clico em Fechar O.C")
	public void fecharOC() {
		// pageCompra.fecharOC();
	}

	@And("quando aparecer alert eu confirmo a compra")
	public void confirmarCompra() {
		// pageCompra.confirmarCompra();
	}
}

File: TestesAuto/BildPython/004_Compra/features/steps/compra_steps.py
from behave import *
from time import sleep

from ..pages.compra_page import compra_page


compra_page = compra_page()

@Given ('Acesso o site "{site}"')
def acessar_site(context, site):
	compra_page.acessar_site(site)

@When ('Realizo login com email "{email}" e senha "{senha}"')
def logar(context, email, senha):
	compra_page.logar(email, senha)

@Then ('vou para home')
def home(context):
	compra_page.home()

@Then('acessei o menu Compras >> Calendario de Compras')
def acessa_menu_compras(context):
	compra_page.acessa_menu_compras()

@Then('acesso calendario')
def acessa_calendario(context):
	compra_page.acessa_calendario()

@Then('na tela de Comprar Insumos seleciono a obra "{obra}"')
def seleciono_obra(context, obra):
	compra_page.seleciona_obra(obra)

@Then('na tela de Comprar Insumos clico em comprar')
def clicar_comprar(context):
	compra_page.clica_comprar()

@Then('na tela de insumos clico em comprar saldo de todos insumos')
def comprar_saldo(context):
	compra_page.comprar_saldo()

@Then('na tela de oc clico em fechar oc')
def fechar_oc(context):
	compra_page.fechar_oc()

@Then('no PopUp de confirmação clico em sim')
def confirmar(context):
	compra_page.confirma()